### PR TITLE
ASoC: SOF: mediatek: Revise mt8195 boot flow

### PR DESCRIPTION
**Pull request by yaochunhung**
Hi,
This patch will fix statVectorSel wrong setting and modify shutdown flow to pull dsp runstall high before reset dsp to avoid unexpected error.
Pleas help to review it. Thanks.

### CODE REVIEW

**Comment by yaochunhung** [comment]
SOFCI TEST

File: sound/soc/sof/mediatek/mt8195/mt8195-loader.c [hunk 1]
```diff
@@ -21,14 +21,17 @@ void sof_hifixdsp_boot_sequence(struct snd_sof_dev *sdev, u32 boot_addr)
 
 	/* pull high StatVectorSel to use AltResetVec (set bit4 to 1) */
 	snd_sof_dsp_update_bits(sdev, DSP_REG_BAR, DSP_RESET_SW,
-				DSP_RESET_SW, DSP_RESET_SW);
+				STATVECTOR_SEL, STATVECTOR_SEL);
```
Comment (ujfalusi) [inline sound/soc/sof/mediatek/mt8195/mt8195-loader.c:24]: This change should be a separate one with a Fixes: tag?

**Review by ujfalusi — CHANGES_REQUESTED** [review]
@yaochunhung, I would separate the patch to a fix (STATVECTOR_SEL correction) and the rest.

File: sound/soc/sof/mediatek/mt8195/mt8195-loader.c (hunk 1 above)
Comment (yaochunhung) [reply sound/soc/sof/mediatek/mt8195/mt8195-loader.c:24]: OK. I will separate the patches and aslo use Fixes : tag for STATVECTOR_SEL. thanks.

**Comment by yaochunhung** [comment]
SOFCI TEST

**Comment by yaochunhung** [comment]
Hi all,
Could someone help to point out what happens for this test failure? I wonder why this PR cause this failure. Thanks.

**Comment by bardliao** [comment]
> Hi all, Could someone help to point out what happens for this test failure? I wonder why this PR cause this failure. Thanks.

The check-suspend-resume-with-playback-5.sh and check-suspend-resume-with-capture-5.sh failures of TGLU_RVP_SDW is a known issue https://github.com/thesofproject/linux/issues/3459

File: sound/soc/sof/mediatek/mt8195/mt8195-loader.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/sof/mediatek/mt8195/mt8195-loader.c:24]: you can use a fixup! prefix if the code you are trying to fix has not landed in Mark Brown branch, or you need to provide a Fixes tag.
Which is it?

Comment (yaochunhung) [reply sound/soc/sof/mediatek/mt8195/mt8195-loader.c:24]: Sorry for my misunderstanding, it should be Fixes: tag and I will modify it. Thanks.

**Comment by yaochunhung** [comment]
Can someone help to review this PR? Thanks a lot.

**Review by afq984 — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Review by ranj063 — APPROVED** [review]

**Comment by plbossart** [comment]
@ujfalusi you good with this PR?

**Comment by yaochunhung** [comment]
@ujfalusi  Could you please help to review this PR? Thanks.

**Review by kuanhsuncheng — APPROVED** [review]

**Comment by plbossart** [comment]
@ujfalusi ping. You were the only reviewer requesting changes.

**Comment by kuanhsuncheng** [comment]
Hi, Everyone, are we all good with this PR?

**Review by ujfalusi — APPROVED** [review]

**Pull request merged**